packages/remix-server-runtime/server.ts

```diff
@@ -49,6 +49,15 @@ export const createRequestHandler: CreateRequestHandlerFunction = (
       );
     }
 
+    let overrideMethod = url.searchParams.get("_method") ?? "";
+    if (["put", "patch", "delete"].includes(overrideMethod)) {
+      url.searchParams.delete("_method");
+      request = new Request(url.toString(), {
+        ...request,
+        method: overrideMethod,
+      });
```

Comment:
I like the ideas in general. But this might not work as expected.
Check this [StackBlitz](https://stackblitz.com/edit/js-rfla3u?devToolsHeight=33&file=index.js). 

What if we keep the `_method` on the request url similar to how remix handles `_data` currently and just override the method?

```ts
request = new Request(request, {
    method: overrideMethod,
});
```

Reply:
Oh yikes, good catch! I'll tweak this 👍

Reply:
Fixed ... expanded the tests in the process, which uncovered another issue 😅. Also moved the server logic to live alongside related stuff in data.ts